libewf/libewf_single_files.c

```diff
@@ -988,50 +1533,2314 @@ int libewf_single_files_parse_record_values(
+			"%s: unable to retrieve number of values",
+			function);
+
+		goto on_error;
+	}
+#if defined( HAVE_VERBOSE_OUTPUT )
+	if (number_of_types != number_of_values)
+	{
+		if (libcnotify_verbose != 0)
+		{
+			libcnotify_printf(
+				"%s: mismatch in number of types and values.\n",
+				function);
+		}
+	}
+#endif
+	if (libewf_single_file_source_initialize(
+		&single_file_source,
+		error) != 1)
+	{
+		libcerror_error_set(
+			error,
+			LIBCERROR_ERROR_DOMAIN_RUNTIME,
+			LIBCERROR_RUNTIME_ERROR_INITIALIZE_FAILED,
+			"%s: unable to create single file source.",
+			function);
+
+		goto on_error;
+	}
+	for (value_index = 0;
+		value_index < number_of_types;
+		value_index++)
+	{
+		if (libfvalue_split_utf8_string_get_segment_by_index(
+			types,
+			value_index,
+			&type_string,
+			&type_string_size,
+			error) != 1)
+		{
+			libcerror_error_set(
+				error,
+				LIBCERROR_ERROR_DOMAIN_RUNTIME,
+				LIBCERROR_RUNTIME_ERROR_GET_FAILED,
+				"%s: unable to retrieve type string: %d.",
+				function,
+				value_index);
+
+			goto on_error;
+		}
+		if ((type_string == NULL)
+			|| (type_string_size < 2)
+			|| (type_string[0] == 0))
+		{
+			libcerror_error_set(
+				error,
+				LIBCERROR_ERROR_DOMAIN_RUNTIME,
+				LIBCERROR_RUNTIME_ERROR_VALUE_MISSING,
+				"%s: missing type string: %d.",
+				function,
+				value_index);
+
+			goto on_error;
+		}
+		/* Remove trailing carriage return
+		 */
+		else if (type_string[type_string_size - 2] == (uint8_t) '\r')
+		{
+			type_string[type_string_size - 2] = 0;
+
+			type_string_size -= 1;
+		}
+		if (value_index < number_of_values)
+		{
+			if (libfvalue_split_utf8_string_get_segment_by_index(
+				values,
+				value_index,
+				&value_string,
+				&value_string_size,
+				error) != 1)
+			{
+				libcerror_error_set(
+					error,
+					LIBCERROR_ERROR_DOMAIN_RUNTIME,
+					LIBCERROR_RUNTIME_ERROR_GET_FAILED,
+					"%s: unable to retrieve value string: %d.",
+					function,
+					value_index);
+
+				goto on_error;
+			}
+			if ((value_string == NULL)
+				|| (value_string_size < 2)
+				|| (value_string[0] == 0))
+			{
+				value_string = NULL;
+				value_string_size = 0;
+			}
+			/* Remove trailing carriage return
+			 */
+			else if (value_string[value_string_size - 2] == (uint8_t) '\r')
+			{
+				value_string[value_string_size - 2] = 0;
+
+				value_string_size -= 1;
+			}
+		}
+		else
+		{
+			value_string = NULL;
+			value_string_size = 0;
+		}
+#if defined( HAVE_DEBUG_OUTPUT )
+		if (libcnotify_verbose != 0)
+		{
+			libcnotify_printf(
+				"%s: type: %s with value: %s.\n",
+				function,
+				(char *)type_string,
+				(char *)value_string);
+		}
+#endif
+		if (value_string == NULL)
+		{
+			/* Ignore empty values
+			 */
+		}
+		else if (type_string_size == 4)
+		{
+			if ((type_string[0] == (uint8_t) 'p')
+			 && (type_string[1] == (uint8_t) 'g')
+			 && (type_string[2] == (uint8_t) 'u'))
+			{
+				single_file_source->primary_guid = (uint8_t *)memory_allocate(
```

Comment:
@jtsylve are these GUIDs little or big endian? do you also have a sample file / sample section that contains a source.

Reply:
I found a sample file, but if you have closure on the byte order let me know.